src/CodeGen_LLVM.cpp

```diff
@@ -1301,6 +1302,19 @@ Expr CodeGen_LLVM::sorted_avg(Expr a, Expr b) {
     return a + (b - a)/2;
 }
 
+// Although these look like intrinsics, they are designed to be
```

Comment:
I actually attempted to do exactly that before (move integer division optimization to a lowering pass). It's messy because it doesn't play nicely with other lowering passes. Basically, this would *need* to be one of the last passes, and definitely after simplification.

I suppose maybe an alternative would be to have a "lowering pass" that only manipulates division exprs. So, the implementation of visit(const Div*) for a CodeGen class could just be:

    codegen(implement_division(op));

Which would rewrite division as div_round_to_zero and the other ops.

I'll try doing that now.